### DISCUSSION TITLE
How to use Authentication in Swift

### DISCUSSION THREAD

**Discussion by iankoex** [supabase/supabase#1400] [Questions] [answered]
I understand that there is currently no client library for swift. Is there another way i can use Supabase with Swift? I would really love to use Supabase.

**Comment by matthewmorek** [comment, marked as answer]
@iankoex There's a community-maintained [Supabase Swift SDK](https://github.com/supabase-community/supabase-swift), and although the docs are sparse, you could authenticate with email and password this way using async/await:

```swift
import Foundation
import Supabase

struct SupabaseAPI {
    lazy private var client = SupabaseClient(supabaseURL: apiUrl, supabaseKey: apiKey)

    mutating func signIn(with email: String, and password: String) async throws {
        guard let session = try? await client.auth.signIn(email: email, password: password) else {
            throw APIError.invalidCredentials
        }
    }
}
